azurerm/internal/services/compute/availability_set_resource.go

```diff
@@ -139,12 +153,11 @@ func resourceArmAvailabilitySetCreateUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
-	resp, err := client.CreateOrUpdate(ctx, resGroup, name, availSet)
-	if err != nil {
-		return err
+	if _, err := client.CreateOrUpdate(ctx, resGroup, name, availSet); err != nil {
+		return fmt.Errorf("creating/updating Availability Set %q (Resource Group %q): %+v", name, resGroup, err)
 	}
 
-	d.SetId(*resp.ID)
+	d.SetId(availabilitySetId.ID(subscriptionId))
```

Comment:
ordinarily we'd need to ensure this is rewritten in the Read functions in downstream resources, that is:

- azurerm_linux_virtual_machine
- azurerm_site_recovery_replicated_vm
- azurerm_windows_virtual_machine

else users will have a diff here - however in this instance all of these are ignoring the casing on the Resource ID, due to the API returning this incorrectly - so this _appears_ fine